### PR TITLE
[CN-1251]: reduce init containers to one

### PR DESCRIPTION
**Pull request by kutluhanmetin**
## Description

This PR includes changes that decreases the number of init containers to one.

### CODE REVIEW

File: internal/controller/hazelcast/hazelcast_resources.go [hunk 1]
```diff
@@ -2565,9 +2565,9 @@ func containerSecurityContext() *v1.SecurityContext {
 
 func initContainer(h *hazelcastv1alpha1.Hazelcast, pvcName string) (v1.Container, error) {
 	c := corev1.Container{
-		Name:    n.InitContainer,
-		Image:   h.AgentDockerImage(),
-		Command: []string{"execute-multiple-commands"},
+		Name:  n.InitContainer,
+		Image: h.AgentDockerImage(),
+		Args:  []string{"execute-multiple-commands"},
```
Comment (SeriyBg) [inline internal/controller/hazelcast/hazelcast_resources.go:2570]: What is the reason for this change?

File: internal/controller/hazelcast/hazelcast_resources.go [hunk 2]
```diff
@@ -2578,6 +2578,16 @@ func initContainer(h *hazelcastv1alpha1.Hazelcast, pvcName string) (v1.Container
 		TerminationMessagePolicy: "File",
 		SecurityContext:          containerSecurityContext(),
 		ImagePullPolicy:          corev1.PullIfNotPresent,
+		VolumeMounts: []v1.VolumeMount{
+			{
+				Name:      pvcName,
+				MountPath: n.PersistenceMountPath,
```
Comment (SeriyBg) [inline internal/controller/hazelcast/hazelcast_resources.go:2584]: Shouldn't it be present only of the persistence is enabled?

File: internal/controller/hazelcast/hazelcast_resources.go [hunk 3]
```diff
@@ -2649,6 +2653,17 @@ func volumes(h *hazelcastv1alpha1.Hazelcast) []v1.Volume {
 				},
 			},
 		},
+		{
+			Name: n.AgentConfigMap,
```
Comment (SeriyBg) [inline internal/controller/hazelcast/hazelcast_resources.go:2657]: Same question here in relation to UCN

File: internal/controller/hazelcast/hazelcast_resources.go [hunk 4]
```diff
@@ -943,33 +942,181 @@ func (r *HazelcastReconciler) reconcileMTLSSecret(ctx context.Context, h *hazelc
...
-	cfgW := compound.ConfigWrapper{}
-	if len(ns) != 0 {
-		buckets := make([]downloadbucket.Cmd, 0)
-		for _, ucn := range ns {
-			buckets = append(buckets, downloadbucket.Cmd{
-				BucketURI:   ucn.Spec.BucketConfiguration.BucketURI,
-				SecretName:  ucn.Spec.BucketConfiguration.SecretName,
-				Destination: filepath.Join(n.UCNBucketPath, ucn.Name+".zip"),
-			})
+	// UCN
```
[the remark this answers is not in the document]
Comment (kutluhanmetin) [reply internal/controller/hazelcast/hazelcast_resources.go]: They just help me refactoring, removed them: https://github.com/hazelcast/hazelcast-platform-operator/pull/1157/commits/c5735cf64360f63683fdda593147af0f2c7e4092

**Review by SeriyBg — APPROVED** [review]

**Review by semihbkgr — APPROVED** [review]
LGTM 👍
